routing/datasource.go

```diff
@@ -0,0 +1,197 @@
+package routing
+
+import (
+	"fmt"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"log"
+	"net/url"
+	"time"
+)
+
+type incomingType uint
+
+const (
+	incomingReset incomingType = iota
+	incomingUpdate
+)
+
+type routeDefs map[string]*eskip.Route
+
+type incomingData struct {
+	typ            incomingType
+	client         DataClient
+	upsertedRoutes []*eskip.Route
+	deletedIds     []string
+}
+
+func receiveInitial(c DataClient, pollTimeout time.Duration, out chan<- *incomingData) {
+	for {
+		routes, err := c.GetInitial()
+		if err != nil {
+			log.Println("error while receiveing initial data;", err)
+			time.Sleep(pollTimeout)
+			continue
+		}
+
+		out <- &incomingData{incomingReset, c, routes, nil}
```

Comment:
always send a reset from a new client? isn't making it hard to use multiple clients in parallel?